core/services/relay/evm/cap_encoder.go

```diff
@@ -0,0 +1,65 @@
+package evm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+const abiConfigFieldName = "abi"
+
+// TODO: include from chainlink-common
+type Encoder interface {
+	Encode(ctx context.Context, input values.Map) ([]byte, error)
+}
+
+type capEncoder struct {
```

Comment:
@bolekk I'm thinking about how this will connect to the capability -- how were you planning on exposing this to the capability?

Reply:
resolved offline